access-common/src/main/java/edu/unc/lib/dl/ui/service/SolrQueryLayerService.java

```diff
@@ -1014,26 +1017,24 @@ public void setCollectionsPid(PID collectionsPid) {
 	 * @return the count, or -1 if there was an error retrieving the count
 	 */
 
-	public int getDepartmentsCount() {
-
-		SolrQuery query;
-		QueryResponse response;
-
-		query = new SolrQuery();
-		query.setQuery("*:*");
-		query.setRows(0);
-		query.addFacetField("department");
-		query.setFacetLimit(-1);
-
+	public int getDepartmentsCount(AccessGroupSet accessGroups) {
 		try {
-			response = this.executeQuery(query);
+			StringBuilder queryBuilder = new StringBuilder("*:*");
+			addAccessRestrictions(queryBuilder, accessGroups);
+
+			SolrQuery query = new SolrQuery();
+			query.setQuery(queryBuilder.toString());
+			query.setRows(0);
+			query.addFacetField("department");
+			query.setFacetLimit(-1);
```

Comment:
what does -1 signal here? i don't see a good explanation of this in the solr docs

Reply:
It means unlimited, the default is some rather low number